### PR TITLE
fix #2382 issue with defaultValues inside submit values

### LINKED ISSUE #2382: [useFieldArray] The value will be the default value when removed all fields
**Describe the bug**
The value will be the default value when removed all fields. If the arguments of useForm has defaultValues.

**To Reproduce**
1. Go to the codesandbox provided below
2. Remove all array fields
3. Click submit
4. See console output

**Codesandbox link (Required)**
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-cqvfs

**Expected behavior**
The data of the submission should be an empty array.

**Desktop:**
 - OS: macOS
 - Browser: chrome
 - Version: 84.0.4147.89

### PR DESCRIPTION
**Pull request by bluebill1049**
@keiya01 this is what i meant. (feel free to take over whether you think this is right or wrong solution)

### CODE REVIEW

**Comment by keiya01** [comment]
Thank you. It looks good to me.
I was worried data of `getValues()` and `handleSubmit()` have no consistent. But I noticed this change has no problem because we can do opt-in `defaultValues` by using `getValues()`.

File: cypress/integration/useFieldArray.ts [hunk 1]
```diff
@@ -170,9 +170,7 @@ describe('useFieldArray', () => {
     cy.get('ul > li').should('have.length', 0);
 
     cy.get('#submit').click();
-    cy.get('#result').contains(
-      '{"data":[{"name":"test"},{"name":"test1"},{"name":"test2"}]}',
-    );
+    cy.get('#result').contains('{}');
```
Comment (bluebill1049) [inline cypress/integration/useFieldArray.ts:173]: can't believe the mistake stays there for long.

**Review by kotarella1110 — APPROVED** [review]
Awesome!
I think in the docs, clarifying the difference between handleSubmit and getValues ​​about defaultValues ​​will avoid user confusion.
What do you think?

**Comment by bluebill1049** [comment]
will do @kotarella1110

**Comment by bluebill1049** [comment]
Screen Shot 2020-07-28 at 5 30 00 pm

**Pull request merged**